ext/webstorage/01_webstorage.js

```diff
@@ -83,51 +83,54 @@ function createStorage(persistent) {
 
   const proxy = new Proxy(storage, {
     deleteProperty(target, key) {
-      if (typeof key == "symbol") {
-        delete target[key];
-      } else {
-        target.removeItem(key);
+      if (typeof key === "symbol") {
+        return ReflectDeleteProperty(target, key);
       }
+      target.removeItem(key);
       return true;
     },
+
     defineProperty(target, key, descriptor) {
-      if (typeof key == "symbol") {
-        ObjectDefineProperty(target, key, descriptor);
-      } else {
-        target.setItem(key, descriptor.value);
+      if (typeof key === "symbol") {
+        return ReflectDefineProperty(target, key, descriptor);
       }
+      target.setItem(key, descriptor.value);
       return true;
     },
-    get(target, key) {
-      if (typeof key == "symbol") return target[key];
+
+    get(target, key, receiver) {
+      if (typeof key === "symbol") {
+        return target[key];
+      }
       if (ReflectHas(target, key)) {
-        return ReflectGet(...new SafeArrayIterator(arguments));
-      } else {
-        return target.getItem(key) ?? undefined;
+        return ReflectGet(target, key, receiver);
       }
+      return target.getItem(key) ?? undefined;
     },
+
     set(target, key, value) {
-      if (typeof key == "symbol") {
-        ObjectDefineProperty(target, key, {
+      if (typeof key === "symbol") {
+        return ReflectDefineProperty(target, key, {
```

Comment:
`defineProperty` is used here, should it be `Reflect.set`?

Reply:
if my memory serves me right, there were some very weird behavioural differences that caused issues, but I might be wrong.